server/src/main/java/com/flipkart/varadhi/auth/DefaultAuthorizationProvider.java

```diff
@@ -0,0 +1,92 @@
+package com.flipkart.varadhi.auth;
+
+import com.flipkart.varadhi.entities.UserContext;
+import com.flipkart.varadhi.exceptions.InvalidConfigException;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.flipkart.varadhi.auth.ResourceAction.ORG_CREATE;
+
+public class DefaultAuthorizationProvider implements AuthorizationProvider {
+    private DefaultAuthorizationConfiguration configuration;
+
+    @Override
+    public Future<Boolean> init(JsonObject configuration) {
+        this.configuration = configuration.mapTo(DefaultAuthorizationConfiguration.class);
+        return Future.succeededFuture(true);
+    }
+
+    /**
+     * Checks if the user {@code UserContext} is authorized to perform action {@code ResourceAction} on resource path.
+     *
+     * @param userContext information on the user, contains the user identifier
+     * @param action      the action being performed by the user which is to be authorized
+     * @param resource    the path of the resource on which the action is to be authorized
+     *
+     * @return {@code Future<Boolean>} a future result expressing True/False decision
+     */
+    @Override
+    public Future<Boolean> isAuthorized(UserContext userContext, ResourceAction action, String resource) {
```

Comment:
Do we need this to be Async ?

Reply:
The current authorization provider interface returns Future.
I think thats so that our handler is not blocking. https://vertx.io/docs/vertx-core/java/#golden_rule